hpat/compiler.py

```diff
@@ -150,145 +63,104 @@ def inline_calls(func_ir, _locals):
-            pm.add_stage(self.stage_parfor_pass, "convert to parfors")
-        pm.add_stage(self.stage_distributed_pass, "convert to distributed")
-        pm.add_stage(self.stage_ir_legalization,
-                     "ensure IR is legal prior to lowering")
-        self.add_lowering_stage(pm)
-        self.add_cleanup_stage(pm)
+        pm = DefaultPassBuilder.define_nopython_pipeline(self.state)
 
-    def stage_inline_pass(self):
-        """
-        Inline function calls (to enable distributed pass analysis)
-        """
-        # Ensure we have an IR and type information.
-        assert self.func_ir
-        inline_calls(self.func_ir, self.locals)
+        add_pass_before(pm, InlinePass, InlineClosureLikes)
+        pm.add_pass_after(HiFramesPass, InlinePass)
+        pm.add_pass_after(DataFramePass, AnnotateTypes)
+        pm.add_pass_after(PostprocessorPass, AnnotateTypes)
+        pm.add_pass_after(HiFramesTypedPass, DataFramePass)
+        pm.add_pass_after(DistributedPass, ParforPass)
+        pm.finalize()
 
-    def stage_df_pass(self):
-        """
-        Convert DataFrame calls
-        """
-        # Ensure we have an IR and type information.
-        assert self.func_ir
-        df_pass = HiFrames(self.func_ir, self.typingctx,
-                           self.args, self.locals, self.metadata)
-        df_pass.run()
+        return [pm]
 
-    def stage_io_pass(self):
-        """
-        Convert IO calls
-        """
-        # Ensure we have an IR and type information.
-        assert self.func_ir
-        if config._has_h5py:
-            io_pass = pio.PIO(self.func_ir, self.locals)
-            io_pass.run()
 
-    def stage_repeat_inline_closure(self):
-        assert self.func_ir
-        inline_pass = InlineClosureCallPass(
-            self.func_ir, self.flags.auto_parallel, typed=True)
-        inline_pass.run()
-        post_proc = postproc.PostProcessor(self.func_ir)
-        post_proc.run()
-
-    def stage_distributed_pass(self):
-        """
-        parallelize for distributed-memory
-        """
-        # Ensure we have an IR and type information.
-        assert self.func_ir
-        from hpat.distributed import DistributedPass
-        dist_pass = DistributedPass(
-            self.func_ir, self.typingctx, self.targetctx,
-            self.type_annotation.typemap, self.type_annotation.calltypes,
-            self.metadata)
-        dist_pass.run()
+@register_pass(mutates_CFG=True, analysis_only=False)
+class ParforSeqPass(FunctionPass):
+    _name = "hpat_parfor_seq_pass"
 
-    def stage_df_typed_pass(self):
-        """
-        Convert HiFrames after typing
-        """
-        # Ensure we have an IR and type information.
-        assert self.func_ir
-        df_pass = HiFramesTyped(self.func_ir, self.typingctx,
-                                self.type_annotation.typemap,
-                                self.type_annotation.calltypes)
-        df_pass.run()
+    def __init__(self):
+        pass
```

Comment:
Looks like it should work the same way without the constructor, constructor of the parent class will be invoked anyway.

Reply:
```__init__``` is marked as an abstract method in the base class